lib/plugins/rotate_asg_instances/asg.rb

```diff
@@ -1,20 +1,35 @@
-require 'moonshot/ssh_fork_executor'
-
 module Moonshot
   class ASG # rubocop:disable Metrics/ClassLength
     include Moonshot::CredsHelper
 
-    def initialize(resources)
+    def initialize(resources, ssh_user)
       @resources = resources
       @ilog = @resources.ilog
-      @ssh_user = ENV['MOONSHOT_SSH_USER'] || ENV['LOGNAME']
+      @ssh_user = ssh_user
+      puts "ssh_user: #{@ssh_user}"
     end
 
     def asg
       @asg ||=
         Aws::AutoScaling::AutoScalingGroup.new(name: physical_resource_id)
     end
 
+    def try_to_ssh(raise_exception: false)
```

Comment:
This method returns true if successful but does not return false when ssh fails(by default)

Reply:
You could do.
```ruby
ssh_executed_successfully?.tap |success|
   raise "XXXXX" unless success
end
```

Reply:
I implemented differently. Please let me know if the approach is acceptable.